source/common/network/socket_impl.cc

```diff
@@ -11,12 +11,24 @@
 namespace Envoy {
 namespace Network {
 
+static IoHandlePtr ioHandleForAddr(Socket::Type type, const Address::InstanceConstSharedPtr addr) {
+  auto sock_interface_name = addr->socketInterface();
+  if (!sock_interface_name.empty()) {
+    auto sock_interface = const_cast<SocketInterface*>(socketInterface(sock_interface_name));
+    if (sock_interface != nullptr) {
+      return sock_interface->socket(type, addr);
+    }
+    return nullptr;
```

Comment:
This will cause a SIGSEGV sometime later, right? Maybe the right thing to do is to RELEASE_ASSERT here that the factory for creating IO handles is missing.

Reply:
Fair point! Made it a RELEASE_ASSERT.